src/Workspaces/Core/Portable/Execution/AssetStorages.Storage.cs

```diff
@@ -26,36 +30,32 @@ public sealed class Storage
             private readonly ISerializerService _serializer;
 
             // additional assets that is not part of solution but added explicitly
-            private ConcurrentDictionary<Checksum, CustomAsset> _additionalAssets;
+            private ConcurrentDictionary<Checksum, CustomAsset>? _lazyAdditionalAssets;
+
+            public SolutionState SolutionState { get; }
```

Comment:
📝 This pattern (property placement before constructor) conflicts with other patterns in active use. We should discuss at the next design meeting and settle on one.

Reply:
If i'm not mistaken the general rule is to declare all members that contribute to the state of the object (data members) upfront, before the ctor declaration. Auto-properties, unlike properties with explicit accessors, are data members. Hence they belong here.

Reply:
See https://github.com/dotnet/roslyn/pull/40091/files#diff-9914e58bcb07148a02f13b2cd9368212L23

Reply:
I have seen other recent pull requests move properties from this position back below the constructor, hence my comment.

Reply:
Yeah, we should settle on the above approach.